File: yacm_ws/src/main/java/jp/ac/yasunagalab/yacm_ws/config/LdapConfig.java
package jp.ac.yasunagalab.yacm_ws.config;


import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@ConfigurationProperties(prefix = "ldap-server")
@Component
@Data
public class LdapConfig {
    private String address;
    private Integer port;


    public String getLdapURL() {
        return "http://" + this.address + ":" + this.port;
    }
}

File: yacm_ws_ldap/src/main/kotlin/jp/ac/kyutech/bio/yasunagalab/yacm_ws_ldap/config/LdapConfig.kt
package jp.ac.kyutech.bio.yasunagalab.yacm_ws_ldap.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConfigurationProperties(prefix = "ldap")
@ConstructorBinding
data class LdapConfig(
    val ldifPath: String
)